### PR TITLE
Refactor release pipeline

### PR DESCRIPTION
**Pull request by nikhil-thomas**
Setup pipelines using workspaces instead of the old PipelineResources based tasks

Refactor release pipeline to accomodate changes in directory structure,
build mechanism etc after knative/pkg based operator refactoring

Remove Task manifests  which are no longer needed

Add Release docs

Accomodate changes based on https://github.com/tektoncd/operator/pull/239

---

**note**: release docs and cheat sheet doc pending
---
Signed-off-by: Nikhil Thomas <nikthoma@redhat.com>

# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [ ] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [ ] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

File: tekton/kustomization.yaml [hunk 1]
```diff
@@ -0,0 +1,5 @@
+apiVersion: kustomize.config.k8s.io/v1beta1
+kind: Kustomization
+resources:
+  - publish-images-manifests.yaml
+  - releasipeline.yaml
```
Comment (savitaashture) [inline tekton/kustomization.yaml:5]: I think these file names should be updated to 
```
build-publish-images-manifests.yaml
operator-release-pipeline.yaml
```
:thinking:

Comment (nikhil-thomas) [reply tekton/kustomization.yaml:5]: true 👍

Comment (nikhil-thomas) [reply tekton/kustomization.yaml]: done

File: tekton/build-publish-images-manifests.yaml [hunk 2]
```diff
@@ -0,0 +1,195 @@
...
+      mkdir -p ${DOCKER_CONFIG}
+      cp /workspace/docker-config.json ${DOCKER_CONFIG}/
+
+      # Change to directory with our .ko.yaml
+      cd ${PROJECT_ROOT}
+
+      # For each cmd/<kubeDistro>/* directory, include a full gzipped tar of all source in
+      # vendor/. This is overkill. Some deps' licenses require the source to be
+      # included in the container image when they're used as a dependency.
+      # Rather than trying to determine which deps have this requirement (an(params.imageRegistryd
```
Comment (vdemeester) [inline tekton/build-publish-images-manifests.yaml:105]: `s/an(params.imageRegistryd/and`

File: tekton/build-publish-images-manifests.yaml [hunk 3]
```diff
@@ -0,0 +1,195 @@
...
+      # will delete after review and full debug nikhil-thomas :)
+      function breakPoint() {
+        waitFileName=${1:-waitFile}
+        while [[ ! -f ${waitFileName} ]]; do
+          sleep 10;
+          echo \*\* --------------------------------------- \*\*
+          echo \*\* breakPoint                              \*\*;
+          echo \*\* run \`touch ${waitFileName}\` to resume \*\*
+        done
+      }
```
Comment (vdemeester) [inline tekton/build-publish-images-manifests.yaml:132]: 😅

**Review by vdemeester — APPROVED** [review]

File: tekton/build-publish-images-manifests.yaml (hunk 2 above)
Comment (nikhil-thomas) [reply tekton/build-publish-images-manifests.yaml:105]: oops..

File: tekton/build-publish-images-manifests.yaml (hunk 3 above)
Comment (nikhil-thomas) [reply tekton/build-publish-images-manifests.yaml:132]: removing :D

File: tekton/README.md [hunk 4]
```diff
@@ -0,0 +1,28 @@
+# Tekton Operator Repo CI/CD
+
+_Why does Tekton pipelines have a folder called `tekton`? Cuz we think it would be cool
```
Comment (vdemeester) [inline tekton/README.md:3]: "Why does Tekton operator …" maybe ?

**Pull request merged**